frontend_tests/node_tests/pm_conversations.js

```diff
@@ -68,3 +70,31 @@ test("insert_recent_private_message", () => {
     ]);
     assert.deepEqual(pmc.recent.get_strings(), ["1", "1,2,3", "15", "3", "1,2"]);
 });
+
+test("muted_users", () => {
+    pmc.recent.initialize(params);
+
+    // Base data
+    assert.deepEqual(pmc.recent.get(), [
+        {user_ids_string: "1", max_message_id: 100},
+        {user_ids_string: "3", max_message_id: 99},
+        {user_ids_string: "1,2", max_message_id: 98},
+        {user_ids_string: "1,2,3", max_message_id: 97},
+        {user_ids_string: "15", max_message_id: 96},
+    ]);
+    assert.deepEqual(pmc.recent.get_strings(), ["1", "3", "1,2", "1,2,3", "15"]);
+
+    // Mute some users.
+    muting.add_muted_user(1);
+    muting.add_muted_user(2);
+
+    // We should now get back only those messages which are either-
+    // 1:1 PMs in which the other user hasn't been muted.
+    // Huddles where there's at least one non-muted participant.
+    assert.deepEqual(pmc.recent.get(), [
+        {user_ids_string: "3", max_message_id: 99},
+        {user_ids_string: "1,2,3", max_message_id: 97},
+        {user_ids_string: "15", max_message_id: 96},
+    ]);
+    assert.deepEqual(pmc.recent.get_strings(), ["3", "1,2,3", "15"]);
```

Comment:
This can be a follow-up, but it'd be nice to have a test here to ensure that "PMs with yourself" remain listed.

Reply:
User ID 15 is indeed the "self" user, so the last message in that array is a PM with yourself, which is listed. Am I missing something?

Reply:
No, we're good, I just hadn't read the test carefully.